tests/visualization_tests/test_intermediate_plot.py

```diff
@@ -69,7 +103,9 @@ def objective(trial: Trial, report_intermediate_values: bool) -> float:
         _IntermediatePlotInfo(trial_infos=[]),
```

Comment:
By adding following example, you'll see the test fail when `python<=3.8`
ref. https://github.com/optuna/optuna/pull/4851#discussion_r1282631566
```
        _IntermediatePlotInfo(
            trial_infos=[
                _TrialInfo(
                    trial_number=0, sorted_intermediate_values=[(0, 1.0), (1, 2.0)], feasible=True
                ),
                _TrialInfo(
                    trial_number=1, sorted_intermediate_values=[(1, 2.0), (0, 1.0)], feasible=False
                ) 
            ]
        )
```